### PR TITLE
Fix issues caused by PJAX

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type
**What kind of change does this PR introduce?**

- [x] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
Good job!

File: scripts/helpers/engine.js [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 'use strict';
 
+const crypto = require('crypto');
+
+hexo.extend.helper.register('gitalk_md5', function(path) {
+  var str = this.url_for(path);
+  str = encodeURI(str);
```
Comment (stevenjoezhang) [inline scripts/helpers/engine.js]: This line should be removed due to https://github.com/hexojs/hexo/pull/3708

`post.path` is encoded twice after Hexo 4.0.0

File: scripts/helpers/engine.js [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+  str.replace('index.html', '');
```
Comment (stevenjoezhang) [inline scripts/helpers/engine.js:10]: This line should be `str = str.replace('index.html', '');`

Comment (stevenjoezhang) [inline scripts/helpers/engine.js:10]: This line should be `str = str.replace('index.html', '');`

**Pull request merged**